constantine/serialization/io_limbs.nim

```diff
@@ -118,8 +127,17 @@ func unmarshalBE[T](
 
     # if full, dump
     if acc_len >= wordBitWidth:
-      if dst_idx == dst.len:
-        return false
+      if dst_idx == dst.len: # return, however check if all remaining
+        # elements are 0 (more significant bits beyond `src_idx-1`)
+        # For examplle, given a
+        # `src = "0000AABBCCDD"`
+        # and a 4 byte BigEndian `BigInt`, `src` is fine, because the most
+        # significant two bytes are all zero.
+        var allZero = true
+        for jidx in countdown(src_idx, 0):
+          if src[jidx] != 0: allZero = false
```

Comment:
same remark, analyzing the data itself instead of metadata (length) breaks constant-time.